app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,231 +29,288 @@ const EMPTY_ACCOUNT = {
     });
 
-    alertController = new AlertController({
-      state: {
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
-      },
-      controllerMessenger: alertMessenger,
-    });
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    jest.fn().mockReturnValue(EMPTY_ACCOUNT),
+  );
+
+  const controller = new AlertController({
+    messenger: alertControllerMessenger,
+    state: getDefaultAlertControllerState(),
+    ...alertControllerOptions,
+  });
+
+  return await fn({
+    controller,
+    messenger: controllerMessenger,
   });
+}
 
+describe('AlertController', () => {
   describe('default state', () => {
-    it('should be same as AlertControllerState initialized', () => {
-      expect(alertController.store.getState()).toStrictEqual({
-        alertEnabledness: {
-          unconnectedAccount: true,
-          web3ShimUsage: true,
-        },
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
+    it('should be same as AlertControllerState initialized', async () => {
+      await withController({ state: initialState }, ({ controller }) => {
+        expect(controller.state).toStrictEqual({
+          alertEnabledness: {
+            unconnectedAccount: true,
+            web3ShimUsage: true,
+          },
+          unconnectedAccountAlertShownOrigins: {
+            testUnconnectedOrigin: false,
+          },
+          web3ShimUsageOrigins: {
+            testWeb3ShimUsageOrigin: 0,
+          },
+        });
       });
     });
   });
 
   describe('alertEnabledness', () => {
-    it('should default unconnectedAccount of alertEnabledness to true', () => {
-      expect(
-        alertController.store.getState().alertEnabledness.unconnectedAccount,
-      ).toStrictEqual(true);
+    it('should default unconnectedAccount of alertEnabledness to true', async () => {
+      await withController({ state: initialState }, ({ controller }) => {
+        expect(
+          controller.state.alertEnabledness.unconnectedAccount,
+        ).toStrictEqual(true);
+      });
     });
 
-    it('should set unconnectedAccount of alertEnabledness to false', () => {
-      alertController.setAlertEnabledness('unconnectedAccount', false);
-      expect(
-        alertController.store.getState().alertEnabledness.unconnectedAccount,
-      ).toStrictEqual(false);
-      expect(
-        controllerMessenger.call('AlertController:getState').alertEnabledness
-          .unconnectedAccount,
-      ).toStrictEqual(false);
+    it('should set unconnectedAccount of alertEnabledness to false', async () => {
+      await withController(
+        { state: initialState },
+        ({ controller, messenger }) => {
```

Comment:
Similar point as above: can we simplify this?

``` suggestion
      await withController(({ controller, messenger }) => {
```

I won't make suggestions on the remaining tests, but the same applies.